src/main/kotlin/io/snyk/plugin/snykcode/core/SnykCodeUtils.kt

```diff
@@ -21,27 +21,29 @@ class SnykCodeUtils private constructor() : DeepCodeUtilsBase(
     fun allProjectFilesCount(project: Project): Int = allProjectFiles(project).size
 
     override fun allProjectFiles(projectO: Any): Collection<Any> {
+        scLogger.logInfo("allProjectFiles requested")
         val project = PDU.toProject(projectO)
-        return RunUtils.computeInReadActionInSmartMode(
+        val progressIndicator = ProgressManager.getInstance().progressIndicator ?: null
+        val allVirtualFiles = mutableSetOf<VirtualFile>()
+
+        val cancelled = !ProjectRootManager.getInstance(project).fileIndex.iterateContent {
```

Comment:
@ArtsiomCh , do we iterate over all project files (included, excluded, ignored etc)? Could  you elaborate the difference between previous and current method to get files?

Reply:
I remember we talked about it.
This is my point: we have now 3 levels where we filter out source files:
- .gitignore
- .dciignore
- IntelliJ excluded/ignored (from gitignore) files/folders.

I guess it's ok and we should use IntelliJ API.